### PR TITLE
Document adding a custom catalog type

### PR DESCRIPTION
**Pull request by adamdmharvey**
## Hey, I just made a Pull Request!

Adds instructions in the documentation for how to extend the catalog model to support a new type.

I tested this in my local installation, and I believe this is the extent of the customization needed, though there may be more I've missed so welcome edits.

Addition of "iOS Applications" custom type:

image

Default EntityPage:

image

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/spotify/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by Fox32** [comment]
Related to #2791 

I guess we don't need to document how to do it in the backstage code, but make it configurable and document how to do it in an own app.

**Comment by freben** [comment]
And also the higher level of how does one use versions, kinds and types, when is it appropriate to put something as a label, annotation or spec field, etc

**Comment by andrewthauer** [comment]
@freben - On that note, I made some updates to #2791, and am looking for some initial feedback.

File: docs/features/software-catalog/extending-the-model.md [hunk 1]
```diff
@@ -22,8 +22,59 @@ the years, and now includes ML models, Apps, data pipelines and many more.
...
+2. Set the new type in your component (e.g., modify your `catalog-info.yaml`)
+
+You can also optionally modify the page to display different information when
+the component is displayed (see
+[`EntityPage.tsx`](https://github.com/spotify/backstage/blob/master/packages/app/src/components/catalog/EntityPage.tsx)),
+but this example will use the `DefaultEntityPage` with the Overview and Docs
+tabs.
+
+For example, to add a new type of iOS Application, first modify the `const tabs`
+variable in `plugins/catalog/src/components/CatalogPage/CatalogPage.tsx` like
```
Comment (Rugvip) [inline docs/features/software-catalog/extending-the-model.md:37]: As mentioned, this isn't a file that you have access to as a user of Backstage. Making this configurable is what #2791 is about

**Comment by adamdmharvey** [comment]
Good feedback throughout, so will close as it'll be addressed differently.

**Pull request closed without merging**